src/dualscreen.tsx

```diff
@@ -387,6 +388,11 @@ export class DualScreen extends KalturaPlayer.core.BasePlugin implements IEngine
     }
   };
 
+  private _getThumbs() {
+    const kalturaCuePointService :any = this._player.getService('kalturaCuepoints');
+    kalturaCuePointService.registerTypes([kalturaCuePointService.CuepointType.SLIDE]);
```

Comment:
```suggestion
    kalturaCuePointService?.registerTypes([kalturaCuePointService.CuepointType.SLIDE]);
```

Reply:
Maybe use "hasService" and warn if not?

Reply:
yes

Reply:
I think hasService is a nice addition to API in general, but for maybe other checks.
In this case we ask for service and if it doesn't exist then we just don't register types.
I would wait a bot with adding more API to services if not necessary until we can have better system to define service dependencies.

Reply:
I think in the future when we do have this dependency management then in any case a plugin can wait for a service and require it only once in the constructor or init phase of the plugin and in this case we can handle lack of service a bit more gracefully.
@RoyBregman this is related to the discussion we had about finding good reasons to have this in place.